bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingFactory.java

```diff
@@ -105,9 +108,13 @@ public static Thing createThing(ThingUID thingUID, Configuration configuration,
         for (ThingHandlerFactory thingHandlerFactory : thingHandlerFactories) {
             if (thingHandlerFactory.supportsThingType(thingTypeUID)) {
                 Thing thing = thingHandlerFactory.createThing(thingTypeUID, configuration, thingUID, bridgeUID);
-                if (properties != null) {
-                    for (String key : properties.keySet()) {
-                        thing.setProperty(key, properties.get(key));
+                if (thing == null) {
+                    logger.error("Thing factory returned null when it reports to support the thing");
```

Comment:
Well, we can, but this won't be the only entry that gets logged. Before this is a message saying it's creating a thing of type X, and after this is a message also printing the thing type doesn't exist...  I just felt there was sufficient logging - especially given this is most likely to occur during development when the dev will be looking....